components/button/button-icon.js

```diff
@@ -174,7 +176,7 @@ class ButtonIcon extends ThemeMixin(ButtonMixin(VisibleOnAncestorMixin(RtlMixin(
 				name="${ifDefined(this.name)}"
 				title="${ifDefined(this.text)}"
 				type="${this._getType()}">
-				<d2l-icon icon="${ifDefined(this.icon)}" class="d2l-button-icon"></d2l-icon>
+				${icon}
```

Comment:
Another option here would be to always render the slot and just provide the icon as the default content. The way slots work is it'll render the default content unless a slot is provided.

```html
<slot name="icon"><d2l-icon icon="${this.icon}" class="d2l-button-icon"></d2l-icon></slot>
```

We'd probably still want to render `nothing` though in the slot if no `icon` attribute is set.